backtrace-library/src/androidTest/java/backtraceio/library/BacktraceClientMetricsTest.java

```diff
@@ -0,0 +1,950 @@
+
+import net.jodah.concurrentunit.Waiter;
+
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+import backtraceio.library.common.BacktraceSerializeHelper;
+import backtraceio.library.common.DeviceAttributesHelper;
+import backtraceio.library.logger.BacktraceLogger;
+import backtraceio.library.logger.LogLevel;
+import backtraceio.library.metrics.BacktraceMetrics;
+import backtraceio.library.metrics.Event;
+import backtraceio.library.metrics.EventsOnServerResponseEventListener;
+import backtraceio.library.metrics.EventsPayload;
+import backtraceio.library.metrics.EventsRequestHandler;
+import backtraceio.library.metrics.EventsResult;
+import backtraceio.library.metrics.UniqueEvent;
+import backtraceio.library.models.BacktraceMetricsSettings;
+import backtraceio.library.models.json.BacktraceAttributes;
+import backtraceio.library.models.types.BacktraceResultStatus;
+
+@RunWith(AndroidJUnit4.class)
+public class BacktraceClientMetricsTest {
+    public Context context;
+    public BacktraceClient backtraceClient;
+    public BacktraceCredentials credentials;
+    private final String summedEventName = "activity-changed";
+    // existing attribute name in Backtrace
+    private final String[] uniqueAttributeName = {"uname.version", "cpu.boottime", "screen.orientation", "battery.state", "device.airplane_mode", "device.sdk", "device.brand", "system.memory.total", "uname.sysname", "application.package"};
+
+    private final String defaultBaseUrl = BacktraceMetrics.defaultBaseUrl;
+    private final String token = "aaaaabbbbbccccf82668682e69f59b38e0a853bed941e08e85f4bf5eb2c5458";
+    private final String universeName = "testing-universe-name";
+    private final String _uniqueEventsSubmissionUrl = "unique-events/submit?token=" + token + "&universe=" + universeName;
+    private final String _summedEventsSubmissionUrl = "summed-events/submit?token=" + token + "&universe=" + universeName;
+
+    static {
+        System.loadLibrary("backtrace-native");
+    }
+
+    /**
+     * NOTE: Some of these tests are very time-sensitive so you may occasionally get false negative results.
+     * For best results run under low CPU load and low memory utilization conditions.
+     */
+
+    @Before
+    public void setUp() {
+        context = InstrumentationRegistry.getContext();
+        credentials = new BacktraceCredentials("https://example-endpoint.com/", "");
+        BacktraceDatabase database = new BacktraceDatabase(context, context.getFilesDir().getAbsolutePath());
+
+        backtraceClient = new BacktraceClient(context, credentials, database);
+
+        BacktraceLogger.setLevel(LogLevel.DEBUG);
+    }
+
+    public class MockRequestHandler implements EventsRequestHandler {
+        public int numAttempts = 0;
+        public int errorCode = 200;
```

Comment:
why this variable name is errorCode if value is 200?

Reply:
Ahh good point, will change to `statusCode`, thank you!